### PR TITLE
Add experimental autocontext api

### LINKED ISSUE #2529: Ilastik API: support autocontext
The title says it all: would be great to use autocontext from the ilastik api (which makes ilastik so much easier to use as a dev! Great work!)

### PR DESCRIPTION
**Pull request by k-dominik**
builds on top of #2859 (first commit here) - so it would be better to review that one first - I'd rebase this PR then.

- [x] Format code and imports.
- [x] Add docstrings and comments.
- [x] Add tests.
- [x] Update [user documentation](https://github.com/ilastik/ilastik.github.io).
- [x] Reference relevant issues and other pull requests.
- [x] Rebase commits into a logical sequence.
- [x] TODO: `ilastik/utility/slottools.py`: why did we `assert not isinstance(dtype, numpy.dtype)`, doesn't really make sense

fixes #2529

### CODE REVIEW

File: tests/test_ilastik/test_experimental/test_api/test_pipelines.py [hunk 1]
```diff
@@ -157,3 +162,60 @@ def test_legacy_operation(self, test_data_lookup: ApiTestDataLookup, input_, pro
...
+        assert prediction_stage_2.shape == expected_prediction_stage_2.shape
+        np.testing.assert_array_almost_equal(prediction_stage_2, expected_prediction_stage_2)
+
+    @pytest.mark.parametrize(
+        "input_, proj",
+        [
+            (TestData.DATA_3_CHANNEL, TestProjects.AUTOCONTEXT_2D),
+        ],
+    )
+    def test_project_wrong_num_channels(self, test_data_lookup, input_, proj):
```
[the remark this answers is not in the document]
Comment (k-dominik) [reply tests/test_ilastik/test_experimental/test_api/test_pipelines.py:198]: What we found over the years is that channel is a special case. Wrong dimensionality means that some axes are missing. Number of channels could still be different even if the channel axis is present.

File: ilastik/experimental/api/_pipelines.py [hunk 2]
```diff
@@ -122,6 +124,134 @@ def get_probabilities(self, raw_data: Union[vigra.VigraArray, xarray.DataArray])
...
+                "Number of spatial dims doesn't match. "
+                f"Classifier trained for {self._num_spatial_dims} but input has {num_spatial_in_data}"
+            )
+
+        self._reorder_op.Input.setValue(raw_data)
+
+        if stage == 1:
+            predict_op = self._predict_op_stage1
+        elif stage == 2:
+            predict_op = self._predict_op_stage2
```
[the remark this answers is not in the document]
Comment (k-dominik) [reply ilastik/experimental/api/_pipelines.py:231]: intersting - are you using pyright or mypy? pyright doesn't need the assurance - but yes, doesn't hurt.

[the remark this answers is not in the document]
Comment (btbest) [reply ilastik/experimental/api/_pipelines.py:232]: I think it might be a custom type checker :o I might end up trying the pyright or mypy plugins...

**Comment by btbest** [comment]
I didn't realise cross-platform differences would cause so much trouble, but I do like this solution more than running headless. Besides the faster test time, we now have an explicit definition of acceptable cross-platform deviation, and I guess this is the first time we have an alert set up if something changes substantially due to changes in our dependencies 🎉

**Pull request merged**